python/dgl/data/chem/csv_dataset.py

```diff
@@ -11,18 +11,51 @@
 from ..utils import download, get_download_dir, _get_dgl_url, Subset
 
 class CSVDataset(object):
+    """CSVDataset
+
+    This is a general class for loading data from csv or pd.DataFrame.
+
+    In data pre-processing, we set non-existing labels to be 0, and returning mask with 1 where label exists.
+
+    All molecules are converted into DGLGraphs. After the first-time construction, the
+    DGLGraphs will be saved for reloading so that we do not need to reconstruct them every time.
+
+    Parameters
+    ----------
+    df: pandas.DataFrame
+    Dataframe including smiles and labels. Can be loaded by pandas.read_csv(file_path).
+    One column includes smiles and other columns for binary labels.
+    Column names other than smiles column would be considered as task names.
+
+    smile2graph: callable, str -> DGLGraph
+    A function turns smiles into a DGLGraph. Default one can be found 
+    at python/dgl/data/chem/utils.py named with smile2graph.
```

Comment:
I think it will be better to update `docs/source/api/python/data.rst` and directly link to `smile2graph` with `:func:`. Make sure you build the doc and take a look at the visual results.

Reply:
Similarly, update `data.rst` for other data related APIs that users should have access to.

Reply:
Would add this later. Let's merge this first.